src/createSchemaForm.tsx

```diff
@@ -430,34 +478,67 @@ export function createTsForm<
       };
       const ctxLabel = meta.description?.label;
       const ctxPlaceholder = meta.description?.placeholder;
-      accum[key] = (
-        <Fragment key={stringKey}>
+
+      return (
+        <Fragment key={prefixedKey}>
           {beforeElement}
           <FieldContextProvider
             control={control}
-            name={stringKey}
+            name={prefixedKey}
             label={ctxLabel}
             placeholder={ctxPlaceholder}
             enumValues={meta.enumValues as string[] | undefined}
             addToCoerceUndefined={addToCoerceUndefined}
             removeFromCoerceUndefined={removeFromCoerceUndefined}
           >
-            <Component key={key} {...mergedProps} />
+            <Component key={prefixedKey} {...mergedProps} />
           </FieldContextProvider>
           {afterElement}
         </Fragment>
       );
-      return accum;
-    }, {} as Record<SchemaKey, React.ReactNode>);
-    const renderedFieldNodes = Object.values(renderedFields);
+    }
+
+    function renderFields(schema: SchemaType, props: PropType<Mapping, SchemaType, PropsMapType> | undefined) {
+      type SchemaKey = keyof z.infer<UnwrapEffects<SchemaType>>;
+      const _schema = unwrapEffects(schema);
+      const shape: Record<string, RTFSupportedZodTypes> = _schema._def.shape();
+      return Object.entries(shape).reduce((accum, [key, type]: [SchemaKey, RTFSupportedZodTypes]) => {
+        // we know this is a string but TS thinks it can be number and symbol so just in case stringify
+        const stringKey = key.toString();
+        accum[stringKey] = renderComponentForSchemaDeep(type, props as any, stringKey, stringKey, getValues()[key]);
+        return accum;
+      }, {} as RenderedObjectElements);
+    }
+
+    const renderedFields = renderFields(schema, props);
+    const renderedFieldNodes = flattenRenderedElements(renderedFields);
     return (
       <FormProvider {..._form}>
-        <ActualFormComponent {...formProps} onSubmit={submitFn} >
+        <ActualFormComponent {...formProps} onSubmit={() => {return submitFn()}} >
```

Comment:
i was using this for debugging. let me know if you want me to pass it directly instead